### ISSUE TITLE
‘d2 fmt’ to modify SPAN

### ISSUE THREAD

**Issue by bo-ku-ra** [terrastruct/d2#2107]
the following is difficult to read, because it is not a correct description.
```d2
shape: sequence_diagram;

a
a.span.z
a.b
a.span.y
```

\>d2 fmt

outputed
```d2
shape: sequence_diagram;

a
a.span.z
a.span.b
a.span.y
```
or
```d2
shape: sequence_diagram;

a
a.span.z
a.span.y
a.b
```

**Comment by bo-ku-ra** [comment]
can d2 description reach mermaid?

quoted from https://mermaid.js.org/syntax/sequenceDiagram.html
```mermaid
sequenceDiagram
    Alice->>Bob: Hello Bob, how are you ?
    Bob->>Alice: Fine, thank you. And you?
    create participant Carl
    Alice->>Carl: Hi Carl!
    create actor D as Donald
    Carl->>D: Hi!
    destroy Carl
    Alice-xCarl: We are too many
    destroy Bob
    Bob->>Alice: I agree
```

**Comment by stappersg** [comment]
FWIW:
* The first comment, the _can d2 description reach mermaid?_,  might worth it's own issue.
* From the issue self, the _‘d2 fmt’ to modify SPAN_, is it unclear what a possible outcome should be.

**Comment by bo-ku-ra** [comment]
sorry, i didn't explain enough.
i thought that the current way of describing d2 might be a dead end against 'd2 fmt' and 'marmeid'.

**Comment by stappersg** [comment]
On Thu, Sep 19, 2024 at 04:57:56AM -0700, bokura wrote:
> sorry, i didn't explain enough.
> i thought that the current way of describing d2 might be a dead end
> against 'd2 fmt' and 'marmeid'.

I think time should spend on getting a good description
on what the outcome of this issue should be.

And I think also that for the D2 project it would a good thing
that this issue gets closed. It will prevent further drain of human
energy on this issue. So the energy can be spent on D2

 
Regards
Geert Stappers


P.S.
The correct spelling is 'mermaid'.
-- 
Silence is hard to parse

**Comment by alixander** [comment]
@bo-ku-ra I don't understand this issue.

You're suggesting `d2 fmt` should fix broken usages of span in sequence diagrams?

If that's the case, I disagree, fmt is just for syntax conforming. It should not error correct

**Comment by stappersg** [comment]
On Fri, Sep 20, 2024 at 05:47:31AM -0700, bokura wrote:
> Closed #2107 as completed.

   <3

**Comment by alixander** [comment]
@stappersg Please refrain from being antagonistic to others here.

> And I think also that for the D2 project it would a good thing that this issue gets closed. It will prevent further drain of human energy on this issue.

This is especially rude coming from someone new to the project.

I've also had to delete your comment on another discussion that was passive-aggressive. You'll be blocked if you continue in this manner.

**Comment by stappersg** [comment]
On Fri, Sep 20, 2024 at 08:28:31AM -0700, Alexander Wang wrote:
>    ....   You'll be blocked if you continue in this manner.

Acknowledge


Regards
Geert Stappers
-- 
Silence is hard to parse
